pkg/cpio/fs_unix.go

```diff
@@ -98,6 +98,10 @@ func linuxModeToFileType(m uint64) (os.FileMode, error) {
 	return 0, fmt.Errorf("Invalid file type %#o", m&modeTypeMask)
 }
 
+// inodes is a  record of inodes we have created. We do not protect it with a
+// mutex as cpio reading is inherently serial.
+var inodes = map[uint64]Record{}
```

Comment:
Didn't we move away from a global for a reason with the Reading part of this? Can we avoid it here?

Reply:
I can't recall, I modeled this on some other structs we have below. I'm not that happy with this either. One option is to include this map in an outer level function and use it as a parameter to CreateFile/CreateFileInRoot? How does that sound?

Reply:
oh hang on there I think I know what you're saying. brb.